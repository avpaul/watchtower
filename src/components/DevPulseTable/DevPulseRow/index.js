import DevPulseRow from './DevPulseRow';

export default DevPulseRow;
