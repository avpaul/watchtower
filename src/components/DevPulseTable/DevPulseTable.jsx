import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import arrayKey from 'weak-key';

import Table from '../TableComponents/Table';
import TableHeader from '../TableComponents/Header';
import Error from '../Error';
import Loader from '../Loader/Loader';
import DevPulseRow from './DevPulseRow';

const DevPulseTable = props => {
  const { ratings, loading } = props;

  const { ErrorMessage } = Error;
  if (ratings.length === 0 && !loading) {
    return (
      <ErrorMessage message="There's currently no ratings for this fellow" />
    );
  }
  const headers = [
    'Week',
    'Quantity',
    'Quality',
    'Initiative',
    'Communication',
    'Professionalism',
    'Integration'
  ];
  let counter = 0;
  return (
    <Fragment>
      <Table>
        <TableHeader headers={headers} />
        {ratings.map(rating => {
          counter += 1;
          return (
            <DevPulseRow
              key={arrayKey(rating)}
              rating={rating}
              counter={counter}
            />
          );
        })}
      </Table>
      {loading && <Loader />}
    </Fragment>
  );
};

DevPulseTable.defaultProps = {
  fellow: {}
};

DevPulseTable.propTypes = {
  ratings: PropTypes.arrayOf(PropTypes.object).isRequired,
  loading: PropTypes.bool.isRequired,
  fellow: PropTypes.shape({})
};

export default DevPulseTable;
