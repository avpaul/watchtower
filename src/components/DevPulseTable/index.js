import DevPulseTable from './DevPulseTable';

export default DevPulseTable;
