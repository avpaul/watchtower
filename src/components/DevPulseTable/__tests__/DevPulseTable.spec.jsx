import React from 'react';
import { shallow } from 'enzyme';
import DevPulseTable from '../DevPulseTable';

describe('tests the DevPulseTable', () => {
  let wrapper;
  let wrapper2;
  let wrapper3;
  beforeEach(() => {
    const props = {
      ratings: [{}],
      loading: false
    };
    const props2 = {
      ratings: [],
      loading: true
    };

    const props3 = {
      ratings: [],
      loading: false
    };
    wrapper = shallow(<DevPulseTable {...props} />);
    wrapper2 = shallow(<DevPulseTable {...props2} />);
    wrapper3 = shallow(<DevPulseTable {...props3} />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly', () => {
    expect(wrapper2).toMatchSnapshot();
  });
  it('renders correctly', () => {
    expect(wrapper3).toMatchSnapshot();
  });
});
