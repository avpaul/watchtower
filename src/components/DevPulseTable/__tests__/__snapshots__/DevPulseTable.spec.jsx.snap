// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the DevPulseTable renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DevPulseTable
    fellow={Object {}}
    loading={false}
    ratings={
        Array [
            Object {},
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Table>
          <TableHeader
                    headers={
                              Array [
                                        "Week",
                                        "Quantity",
                                        "Quality",
                                        "Initiative",
                                        "Communication",
                                        "Professionalism",
                                        "Integration",
                                      ]
                    }
          />
          <DevPulseRow
                    averageRatings={false}
                    rating={Object {}}
          />
</Table>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <TableHeader
              headers={
                            Array [
                                          "Week",
                                          "Quantity",
                                          "Quality",
                                          "Initiative",
                                          "Communication",
                                          "Professionalism",
                                          "Integration",
                                        ]
              }
/>,
            Array [
              <DevPulseRow
                averageRatings={false}
                rating={Object {}}
/>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "headers": Array [
                "Week",
                "Quantity",
                "Quality",
                "Initiative",
                "Communication",
                "Professionalism",
                "Integration",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "weak-key-0",
            "nodeType": "function",
            "props": Object {
              "averageRatings": false,
              "rating": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Table>
            <TableHeader
                        headers={
                                    Array [
                                                "Week",
                                                "Quantity",
                                                "Quality",
                                                "Initiative",
                                                "Communication",
                                                "Professionalism",
                                                "Integration",
                                              ]
                        }
            />
            <DevPulseRow
                        averageRatings={false}
                        rating={Object {}}
            />
</Table>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <TableHeader
                headers={
                                Array [
                                                "Week",
                                                "Quantity",
                                                "Quality",
                                                "Initiative",
                                                "Communication",
                                                "Professionalism",
                                                "Integration",
                                              ]
                }
/>,
              Array [
                <DevPulseRow
                  averageRatings={false}
                  rating={Object {}}
/>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "headers": Array [
                  "Week",
                  "Quantity",
                  "Quality",
                  "Initiative",
                  "Communication",
                  "Professionalism",
                  "Integration",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "weak-key-0",
              "nodeType": "function",
              "props": Object {
                "averageRatings": false,
                "rating": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`tests the DevPulseTable renders correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DevPulseTable
    fellow={Object {}}
    loading={true}
    ratings={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Table>
          <TableHeader
                    headers={
                              Array [
                                        "Week",
                                        "Quantity",
                                        "Quality",
                                        "Initiative",
                                        "Communication",
                                        "Professionalism",
                                        "Integration",
                                      ]
                    }
          />
</Table>,
        <Loader
          size=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <TableHeader
              headers={
                            Array [
                                          "Week",
                                          "Quantity",
                                          "Quality",
                                          "Initiative",
                                          "Communication",
                                          "Professionalism",
                                          "Integration",
                                        ]
              }
/>,
            Array [],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "headers": Array [
                "Week",
                "Quantity",
                "Quality",
                "Initiative",
                "Communication",
                "Professionalism",
                "Integration",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Table>
            <TableHeader
                        headers={
                                    Array [
                                                "Week",
                                                "Quantity",
                                                "Quality",
                                                "Initiative",
                                                "Communication",
                                                "Professionalism",
                                                "Integration",
                                              ]
                        }
            />
</Table>,
          <Loader
            size=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <TableHeader
                headers={
                                Array [
                                                "Week",
                                                "Quantity",
                                                "Quality",
                                                "Initiative",
                                                "Communication",
                                                "Professionalism",
                                                "Integration",
                                              ]
                }
/>,
              Array [],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "headers": Array [
                  "Week",
                  "Quantity",
                  "Quality",
                  "Initiative",
                  "Communication",
                  "Professionalism",
                  "Integration",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`tests the DevPulseTable renders correctly 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DevPulseTable
    fellow={Object {}}
    loading={false}
    ratings={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "customStyle": Object {},
      "message": "There's currently no ratings for this fellow",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "customStyle": Object {},
        "message": "There's currently no ratings for this fellow",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
